src/trans.rs

```diff
@@ -250,20 +322,78 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
                         match *destination {
                             Some((ref lvalue, _)) => {
-                                let (lvalue_index, _) = self.trans_lval(lvalue);
+                                let dest = self.trans_lval(lvalue);
 
                                 if b_fnty == BinaryenNone() {
                                     // The result of the Rust call is put in MIR into a tmp local,
                                     // but the wasm function returns void (like the print externs)
                                     // TODO: what representation should the unit type have in wasm ?
```

Comment:
since then I've tried the code the compiler outputs in the browser, and the validation rules are stricter, so I also think it probably shouldn't have a representation. I manager to change the compiler to ignore its assignments but we'll probably have to go through everywhere and make fns return void, and so on